dipy/nn/cnn_1d_denoising/cnn_1d_denoising.py

```diff
@@ -0,0 +1,312 @@
+        """
+        Splits the input data into random train and test subsets.
+
+        Parameters:
+        -----------
+        x: numpy array, input data.
+        y: numpy array, target data.
+        test_size: float or int, optional (default=None)
+            If float, should be between 0.0 and 1.0 and 
+            represent the proportion of the dataset to include in the test split.
+            If int, represents the absolute number of test samples.
+            If None, the value is set to the complement of the train size.
+            If train_size is also None, it will be set to 0.25.
+        train_size: float or int, optional (default=None)
+            If float, should be between 0.0 and 1.0 and
+            represent the proportion of the dataset to include in the train split.
+            If int, represents the absolute number of train samples.
+            If None, the value is automatically set to the complement of the test size.
+        random_state: int, RandomState instance or None, optional (default=None)
+            Controls the shuffling applied to the data before applying the split.
+            Pass an int for reproducible output across multiple function calls. See Glossary.
+        shuffle: bool, optional (default=True)
+            Whether or not to shuffle the data before splitting.
+            If shuffle=False then stratify must be None.
+        stratify: array-like, optional (default=None)
+            If not None, data is split in a stratified fashion,
+            using this as the class labels. Read more in the User Guide.
+
+        Returns:
+        Tuple of four numpy arrays: x_train, x_test, y_train, y_test.
+        """
+        sz = x.shape
+        if len(sz) == 4:
+            x = np.reshape(x, (sz[0]*sz[1]*sz[2], sz[3]))
+        sz = y.shape
+        if len(sz) == 4:
+            y = np.reshape(y, (sz[0]*sz[1]*sz[2], sz[3]))
+        return train_test_split(x, y, test_size=test_size, train_size=train_size, 
+                                random_state=random_state, shuffle=shuffle,
+                                stratify=stratify)
+
+    def fit(self, x=None, y=None, batch_size=None, epochs=1, verbose=1,
+        callbacks=None, validation_split=0.0, validation_data=None,
+        shuffle=True, initial_epoch=0, steps_per_epoch=None,
+        validation_steps=None, validation_batch_size=None,
+        validation_freq=1, max_queue_size=10, workers=1,
+        use_multiprocessing=False):
+        """
+        Train the model on train dataset.
+
+        The fit method will train the model for a fixed
+        number of epochs (iterations) on a dataset. If given data 
+        is  4D it will convert it into 1D.
+
+        Parameters
+        ----------
+        x_train         : ndarray
+            The x_train is the train dataset(noise-free data)
+        y_train         : ndarray shape=(BatchSize,)
+            The y_train is the labels of the train dataset(noise-added data)
```

Comment:
Maybe "labels" is fine with this added explanation.

Reply:
Ok